### PR TITLE
Make it easy to prompt user for input in ChatFeed

### PR DESCRIPTION
**Pull request by ahuang11**
image

Closes https://github.com/holoviz/panel/issues/7138

Inspired by https://discourse.holoviz.org/t/panel-chat-interface/7505?u=ahuang11

Wondering if the method should be renamed to simply `prompt` or `prompt_user` or `require_input`

https://github.com/user-attachments/assets/1a2ad16a-b833-48e7-b9d2-5e1efbcad193

```python
import panel as pn

pn.extension()


def prompt_callback(form, instance):
    instance.stream(f"I like {form.value} too!", user="Assistant")


async def callback(message, user, instance):
    if "ice cream" in message.lower():
        form = pn.widgets.TextInput(
            name="Favorite Ice Cream", placeholder="Enter your favorite ice cream"
        )
        instance.prompt_input(
            form, prompt_callback, predicate=lambda form: form.value
        )
        return
    return f"I don't care about {message}"


chat = pn.chat.ChatInterface(callback=callback)

chat.show()
```

I think this can also be generalized to allow editing existing ChatMessages.

Would like your feedback on this @MarcSkovMadsen and @philippjfr

### CODE REVIEW

File: panel/chat/feed.py [hunk 1]
```diff
@@ -807,6 +813,62 @@ def add_step(
             self._chat_log.scroll_to_latest()
         return step
 
+    def prompt_input(
+        self,
+        component: Widget | ListPanel,
+        callback: Callable,
+        predicate: Callable | None = None,
+        timeout: int = 120,
+        button_params: dict | None = {},
```
Comment (ahuang11) [inline panel/chat/feed.py:822]:
```suggestion
        button_params: dict | None = None,
```

File: panel/chat/feed.py [hunk 2]
```diff
@@ -807,6 +813,62 @@ def add_step(
...
+            A predicate to evaluate the component's state, e.g. widget has value.
+            If provided, the button will be enabled when the predicate returns True.
+            The predicate should accept the component as an argument.
+        timeout : int
+            The duration in seconds to wait before timing out.
+        button_params : dict | None
+            Additional parameters to pass to the submit button.
+        """
+        async def _prepare_prompt(*_) -> None:
+            if "name" not in button_params:
```
Comment (ahuang11) [inline panel/chat/feed.py:846]:
```suggestion
            button_params = button_params or {}
            if "name" not in button_params:
```

**Comment by ahuang11** [comment]
https://github.com/user-attachments/assets/4e6ca433-5a06-4cad-b378-b74645cf7fc5

File: panel/chat/feed.py [hunk 3]
```diff
@@ -807,6 +813,91 @@ def add_step(
...
+                    submit_button.param.update(**input_timeout_button_params)
+                    form.disabled = True
+
+        # a trick to call async functions while keeping prompt_user sync
+        # since asyncio.run() cannot be called from a running event loop (notebook)
+        watcher = self.param.watch(_prepare_prompt, '_prompt_trigger')
+        try:
+            self.param.trigger("_prompt_trigger")
+        finally:
+            self.param.unwatch(watcher)
```
Comment (philippjfr) [inline panel/chat/feed.py:899]: Try:

```suggestion
        param.parameterized.async_executor(_prepare_prompt)
```

**Review by philippjfr — APPROVED** [review]

**Pull request merged**